autoload/go/guru.vim

```diff
@@ -217,11 +213,17 @@ endfunc
 
 " run_guru runs the given guru argument
 function! s:run_guru(args) abort
+  let old_gopath = $GOPATH
+  let $GOPATH = go#path#Detect()
   if go#util#has_job()
-    return s:async_guru(a:args)
+    let res = s:async_guru(a:args)
   endif
 
-  return s:sync_guru(a:args)
+  let res = s:sync_guru(a:args)
```

Comment:
This line should now be in an else block so that guru isn't called twice when `async_guru` was called.